controller/scheduler/swarm.py

```diff
@@ -48,14 +52,20 @@ def stop(self, name):
         """
         Stop a container
         """
-        self.docker_cli.stop(name)
+        try:
+            self.docker_cli.stop(name)
+        except:
```

Comment:
This should catch a specific error class or classes. Also if we raise a new Exception here, we have lost the original exception information, and [`RuntimeError`](https://docs.python.org/2/library/exceptions.html#exceptions.RuntimeError) is an uninformative subclass that you use when nothing else fits.

It would be better IMHO to re-raise the native `DockerException` type and ensure that the calling code can deal with that, or at least to add the original exception message and type into the new exception.